cartoframes/context.py

```diff
@@ -244,11 +246,33 @@ def read(self, table_name, limit=None, decode_geom=False, shared_user=None):
 
         query = 'COPY {source} TO stdout WITH (FORMAT csv, HEADER true)'.format(source=source)
 
-        result = self.copy_client.copyto_stream(query)
+        result = self._recursive_read(query, retry_times)
         df = pd.read_csv(result)
```

Comment:
How do we know that the postgres schema is translated in a predictable way to the pd.DataFrame schema? In the original cc.query code, since the return type was JSON we assumed that all data types (except date) were translated appropriately. Then we tried to change the date columns: https://github.com/CartoDB/cartoframes/blob/72e6be78d6e8f9d434b8f4289ddc4d1e01c7d1bf/cartoframes/context.py#L858-L860

Reply:
I am not totally sure about it, but in my tests, everything works fine. We are also "translating" dates in `_clean_DataFrame_from_carto` method. 

Are you thinking of a particular data type?

Reply:
Oo, missed that this will be merged into `read-with-copy`, which I haven't taken a look at yet. But looks 👍 from my quick reading.

Reply:
Yep, it is in a weird state, depending on another open PR